tests/integration/solidity_contracts/EFTests/test_sstore.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+import pytest
+from starkware.starknet.testing.contract import StarknetContract
+
+from tests.utils.helpers import (
+    hex_string_to_bytes_array,
+    hex_string_to_uint256,
+    private_key_from_hex,
+)
+
+logger = logging.getLogger()
+
+
+@pytest.mark.asyncio
+@pytest.mark.EF_TEST
+@pytest.mark.SSTORE
+class TestSSTORE:
+    @pytest.mark.skip("TODO: investigate why nonce is still not updated")
+    async def test_InitCollision_d0g0v0_Shanghai(
+        self,
+        deploy_eoa,
+        create_account_with_bytecode_and_storage,
+        kakarot: StarknetContract,
+    ):
+        # Deploy EOA
+        # From https://github.com/ethereum/tests/blob/develop/src/GeneralStateTestsFiller/stSStoreTest/InitCollisionFiller.json#L155C28-L155C92
+        private_key = private_key_from_hex(
+            "45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8"
+        )
+        caller_eoa = await deploy_eoa(private_key)
+
+        # Pre-deploy contract account
+        storage = {"0x01": "0x01"}
+        contract_account = await create_account_with_bytecode_and_storage(
+            bytecode="", storage=storage, caller_eoa=caller_eoa
+        )
+        # https://github.com/ethereum/tests/blob/develop/BlockchainTests/GeneralStateTests/stSStoreTest/InitCollision.json#L685
+        assert (
+            contract_account.evm_contract_address
+            == "0x6295eE1B4F6dD65047762F924Ecd367c17eaBf8f"
+        )
+        storage_initial = (
+            await contract_account.storage(hex_string_to_uint256("0x01")).call()
+        ).result.value
+        assert storage_initial == hex_string_to_uint256("0x01")
+        nonce_initial = (await contract_account.get_nonce().call()).result.nonce
+        assert nonce_initial == 0
+
+        # Send tx
+        _ = await kakarot.eth_send_transaction(
```

Comment:
Before adding the handling for deployment on existing evm accounts, I was getting this error

```Python
starkware.cairo.lang.vm.vm_exceptions.VmException: /Users/fredtupas/Library/Caches/pypoetry/virtualenvs/kakarot-dlGws2yh-py3.9/lib/python3.9/site-packages/starkware/starknet/common/syscalls.cairo:174:5: Error at pc=0:50:
E           Got an exception while executing a hint.
E               %{ syscall_handler.deploy(segments=segments, syscall_ptr=ids.syscall_ptr) %}
E               ^**************************************************************************^
E           Cairo traceback (most recent call last):
E           ./src/kakarot/kakarot.cairo:157:6: (pc=0:19040)
E           func eth_send_transaction{
E                ^******************^
E           ./src/kakarot/kakarot.cairo:170:12: (pc=0:18994)
E               return Kakarot.eth_call(origin, to, gas_limit, gas_price, value, data_len, data);
E                      ^***********************************************************************^
E           /Users/fredtupas/kkrt-labs/kakarot/src/kakarot/library.cairo:367:69: (pc=0:18523)
E                       let (starknet_contract_address, evm_contract_address) = deploy_contract_account(
E                                                                               ^**********************^
E           /Users/fredtupas/kkrt-labs/kakarot/src/kakarot/library.cairo:265:43: (pc=0:18416)
E                   let (starknet_contract_address) = Accounts.create(class_hash, evm_contract_address);
E                                                     ^***********************************************^
E           /Users/fredtupas/kkrt-labs/kakarot/src/kakarot/accounts/library.cairo:81:33: (pc=0:2062)
E                   let (account_address) = deploy_syscall(
E                                           ^*************^
E
E           Traceback (most recent call last):
E             File "/Users/fredtupas/Library/Caches/pypoetry/virtualenvs/kakarot-dlGws2yh-py3.9/lib/python3.9/site-packages/starkware/starknet/common/syscalls.cairo", line 174, in <module>
E               %{ syscall_handler.deploy(segments=segments, syscall_ptr=ids.syscall_ptr) %}
E             File "/Users/fredtupas/Library/Caches/pypoetry/virtualenvs/kakarot-dlGws2yh-py3.9/lib/python3.9/site-packages/starkware/starknet/core/os/deprecated_syscall_handler.py", line 84, in deploy
E               contract_address = self._deploy(syscall_ptr=syscall_ptr)
E             File "/Users/fredtupas/Library/Caches/pypoetry/virtualenvs/kakarot-dlGws2yh-py3.9/lib/python3.9/site-packages/starkware/starknet/core/os/deprecated_syscall_handler.py", line 530, in _deploy
E               self.sync_state.deploy_contract(contract_address=contract_address, class_hash=class_hash)
E             File "/Users/fredtupas/Library/Caches/pypoetry/virtualenvs/kakarot-dlGws2yh-py3.9/lib/python3.9/site-packages/starkware/starknet/business_logic/state/state_api.py", line 194, in deploy_contract
E               stark_assert(
E             File "/Users/fredtupas/Library/Caches/pypoetry/virtualenvs/kakarot-dlGws2yh-py3.9/lib/python3.9/site-packages/starkware/starkware_utils/error_handling.py", line 188, in stark_assert
E               raise StarkException(code=code, message=message)
E           starkware.starkware_utils.error_handling.StarkException: (500, {'code': <StarknetErrorCode.CONTRACT_ADDRESS_UNAVAILABLE: 4>, 'message': 'Requested contract address 0x2eef988b73556e9000ce1466d0ba3697d040f98030601dbbf725d714fcf8fcd is unavailable for deployment.'})

```

Reply:
you already deployed a contract when you used the `create_account_with_bytecode`.
This address is computed using the nonce of the `caller_eoa`, you need to check if it's increased after the `create_account`, but here, one again, I think that nonce is not increased in `starkware.starknet.testing`, cc @bajpai244